### PR TITLE
Add Message#editActionRows

### PR DESCRIPTION
**Pull request by Xirado**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [x] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This pull request adds Message#editActionRows and MessageChannel#editActionRowwsById

### CODE REVIEW

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 1]
```diff
@@ -1056,6 +1056,78 @@ default MessageAction editMessage(@Nonnull MessageEmbed newContent)
...
+     * @throws java.lang.IllegalStateException
+     *         If the message attempting to be edited was not created by the currently logged in account, or
+     *         if any of the passed-in ActionRows is {@code null}
+     *
+     * @return {@link MessageAction MessageAction}
+     *         <br>The {@link net.dv8tion.jda.api.entities.Message Message} with the updated ActionRows
+     */
+    @Nonnull
+    @CheckReturnValue
+    MessageAction editActionRows(@Nonnull Collection<? extends ActionRow> actionRows);
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:1093]: The method should be called `editComponents` and should accept `Collection<? extends ComponentLayout>` just like in MessageChannel.

Comment (Xirado) [reply src/main/java/net/dv8tion/jda/api/entities/Message.java:1093]: @MinnDevelopment MessageActionImpl also accepts `Collection<? extends ActionRow>`

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 2]
```diff
@@ -1056,6 +1057,78 @@ default MessageAction editMessage(@Nonnull MessageEmbed newContent)
...
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#MISSING_PERMISSIONS MISSING_PERMISSIONS}
+     *     <br>The edit was attempted after the account lost {@link net.dv8tion.jda.api.Permission#MESSAGE_WRITE Permission.MESSAGE_WRITE} in
+     *         the {@link net.dv8tion.jda.api.entities.TextChannel TextChannel}.</li>
+     *
+     *     <li>{@link net.dv8tion.jda.api.requests.ErrorResponse#UNKNOWN_MESSAGE UNKNOWN_MESSAGE}
+     *     <br>The edit was attempted after the Message had been deleted.</li>
+     * </ul>
+     *
+     * @param  actionRows
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:1080]: These do not have to be action rows, there will most certainly be more layouts than that. You should only refer to them as component layouts and name action row as an example.

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 3]
```diff
@@ -1056,6 +1057,78 @@ default MessageAction editMessage(@Nonnull MessageEmbed newContent)
     @CheckReturnValue
     MessageAction editMessageEmbeds(@Nonnull Collection<? extends MessageEmbed> embeds);
 
+    /**
+     * Edits this Message's ActionRows to the provided {@link net.dv8tion.jda.api.interactions.components.ActionRow ActionRows}.
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:1061]: Not necessarily action rows

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 4]
```diff
@@ -1056,6 +1057,78 @@ default MessageAction editMessage(@Nonnull MessageEmbed newContent)
...
+     *         the new ActionRows (up to 5)
+     *
+     * @throws java.lang.UnsupportedOperationException
+     *         If this is a system message
+     * @throws java.lang.IllegalStateException
+     *         If the message attempting to be edited was not created by the currently logged in account, or
+     *         if any of the passed-in ActionRows is {@code null}
+     *
+     * @return {@link MessageAction MessageAction}
+     *         <br>The {@link net.dv8tion.jda.api.entities.Message Message} with the updated ActionRows
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:1090]: Same here, your parameter is component layouts

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 5]
```diff
@@ -1056,6 +1057,78 @@ default MessageAction editMessage(@Nonnull MessageEmbed newContent)
...
+     * </ul>
+     *
+     * @param  actionRows
+     *         the new ActionRows (up to 5)
+     *
+     * @throws java.lang.UnsupportedOperationException
+     *         If this is a system message
+     * @throws java.lang.IllegalStateException
+     *         If the message attempting to be edited was not created by the currently logged in account, or
+     *         if any of the passed-in ActionRows is {@code null}
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:1123]: No `@throws` for `IllegalArgumentException` which is thrown by all the `Check.xyz` methods.

File: src/main/java/net/dv8tion/jda/api/entities/Message.java [hunk 6]
```diff
@@ -1056,6 +1057,78 @@ default MessageAction editMessage(@Nonnull MessageEmbed newContent)
...
+     *
+     * @return {@link MessageAction MessageAction}
+     *         <br>The {@link net.dv8tion.jda.api.entities.Message Message} with the updated ActionRows
+     */
+    @Nonnull
+    @CheckReturnValue
+    MessageAction editComponents(@Nonnull ActionRow... actionRows);
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/Message.java:1130]: This should be a default implementation similar to the one in WebhookClient:

https://github.com/DV8FromTheWorld/JDA/blob/382f9c8fa7d6e2ba485d06812f8c31d23eaa3d09/src/main/java/net/dv8tion/jda/api/entities/WebhookClient.java#L759-L763

File: src/main/java/net/dv8tion/jda/api/entities/MessageChannel.java [hunk 7]
```diff
@@ -3257,6 +3259,188 @@ default MessageAction editMessageEmbedsById(@Nonnull String messageId, @Nonnull
...
+     * </ul>
+     *
+     * @param  messageId
+     *         The id referencing the Message that should be edited
+     * @param  newActionRows
+     *         Up to 5 new {@link net.dv8tion.jda.api.interactions.components.ActionRow ActionRows} for the edited message
+     *
+     * @throws IllegalArgumentException
+     *         <ul>
+     *             <li>If provided {@code messageId} is {@code null} or empty.</li>
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/MessageChannel.java:3382]: Your list is not being closed

File: src/main/java/net/dv8tion/jda/api/entities/MessageChannel.java [hunk 8]
```diff
@@ -3257,6 +3259,188 @@ default MessageAction editMessageEmbedsById(@Nonnull String messageId, @Nonnull
...
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/entities/MessageChannel.java:3428]: Unclosed list

File: src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java [hunk 9]
```diff
@@ -844,10 +846,13 @@ default MessageAction retainFiles(@Nonnull Collection<? extends Message.Attachme
      */
     @Nonnull
     @CheckReturnValue
-    default MessageAction setActionRows(@Nonnull Collection<? extends ActionRow> rows)
+    default MessageAction setActionRows(@Nonnull Collection<? extends ComponentLayout> components)
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java:849]: This is a breaking change nobody asked for, revert it.

File: src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java [hunk 10]
```diff
@@ -820,6 +821,22 @@ public MessageAction editMessageEmbeds(@Nonnull Collection<? extends MessageEmbe
         return ((MessageActionImpl) channel.editMessageEmbedsById(getId(), embeds)).withHook(interactionHook);
     }
 
+    @Nonnull
+    @Override
+    public MessageAction editComponents(@Nonnull Collection<? extends ComponentLayout> actionRows)
```
Comment (MinnDevelopment) [inline src/main/java/net/dv8tion/jda/internal/entities/ReceivedMessage.java:826]: This should be overridden in `SystemMessage` as unsupported as well, you can't edit system messages.

**Review by MinnDevelopment — CHANGES_REQUESTED** [review]

File: src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java (hunk 9 above)
Comment (Xirado) [reply src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java:849]: What's the point in accepting ComponentLayouts if MessageAction only accepts ActionRows?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java:849]: A robust interface.

Comment (Xirado) [reply src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java:849]: so `editComponentsById()` in MessageChannel should also accept ActionRows?

Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java:849]: No. Please just do what i asked?

Comment (Xirado) [reply src/main/java/net/dv8tion/jda/api/requests/restaction/MessageAction.java:849]: Yeah but `editComponentyById` accepts ComponentLayout although MessageActionImpl only accepts ActionRows. Should i cast it? Maybe i'm lost but you're confusing me lol

**Pull request closed without merging**